web/src/engine/events/src/legacyEventEmitter.ts

```diff
@@ -0,0 +1,165 @@
+// Most of the typing below is derived from that of EventEmitter, but customized for
+// modeling legacy KMW events.  Including the heavy typing gets us event Intellisense
+// and compile-time errors if and when types don't match expectations.
+
+/**
+ * Can define the set of events as follows:
+ * ```
+ * interface Test extends EventMap {
+ *   'event': (param: {'prop': any}) => boolean;
+ * }
+ * ```
+ *
+ * Each event may have either no parameters or a single parameter of type object.
+ * The type definition of the parameter will be utilized by TS's type-inference engine
+ * for type checking on handlers and for raising the event.
+ *
+ * Note: the `extends EventMap` part is actually important for TS type inference here.
+ */
+export type LegacyEventMap = object;
+
+/**
+ * Matches the name of any single event defined within the specified event-map definition.
+ */
+export type EventNames<T extends LegacyEventMap> = Exclude<keyof T, number | symbol>;
+
+/**
+ * Builds a type-array of the arguments for each named event, indexed by that name.
+ */
+type ArgumentMap<T extends LegacyEventMap> = {
+  [K in Exclude<keyof T, number | symbol>]: T[K] extends (arg: any) => boolean
+    ? Parameters<T[K]>[0]
+    : (
+      T[K] extends Function
+      ? never
+      : T[K]
+    );
+};
+
+/**
+ * Provides the type signature of event listeners able to handle defined events.
+ */
+export type EventListener<
+    T extends LegacyEventMap,
+    K extends EventNames<T>
+  > = (       // argumentMap[eventName] - retrieves the specific parameter typing for the event.
+        args: ArgumentMap<T>[Extract<K, keyof T>]
+      ) => boolean;
+
+/**
+ * Provides fairly strong typing for all legacy KMW events.  Note that all events
+ * assume a handler receiving up to one object, though that object's properties will
+ * vary from event to event.
+ *
+ * Note that the behavior differs from EventEmitter events on a few points:
+ * 1. Event functions are expected to return a boolean value - generally, `true`.
+ *    If 'false' or `undefined` is returned, no further listeners will receive the event.
+ * 2. These events receive up to one parameter, always of an object type.
+ * 3. These events proactively prevent accidental event-handler recursion.  Should an event's
+ *    handler retrigger the event, the newly-triggered event will be prevented entirely.
+ */
+export class LegacyEventEmitter<EventTypes extends LegacyEventMap> {
```

Comment:
This class implements the same old functions from kmwutils.ts.  The actual implementation is 100% unchanged - the changes are only typing annotations used to provide far, _far_ richer typing and type-inference support.

Anything before this point in this file is also present to facilitate the enhanced typing aspects.